icu4j/main/core/src/main/java/com/ibm/icu/text/UnicodeSet.java

```diff
@@ -5191,63 +5199,70 @@ public IntStream codePointStream() {
-     *
-     * <p><b>Warning:</b> For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
-     * Do not alter the {@code UnicodeSet} while iterating.
+     * <p><b>Warnings:</b>
+     * <ul>
+     *   <li>To iterate over the full contents, you have to also iterate over the strings.
+     *   <li>For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
+     *       Do not alter the {@code UnicodeSet} while iterating.
+     * </ul>
      *
      * @return a {@link Stream} of {@code String}
+     *
+     * @draft ICU 76
      */
     public Stream<String> stream() {
         return StreamSupport.stream(spliterator(), false);
     }
 
     /**
-     * Provide for iteration on code points.
-     *
-     * <p>Returns an {@link Iterable} over all the code points, "expanding" the ranges
-     * by iterating on all code points between the {@link EntryRange#codepoint} and
-     * {@link EntryRange#codepointEnd}.
+     * Returns an {@link Iterable} for iteration over all the code points in this set.
      *
-     * <p><b>Warning:</b> This is a convenience method, but comes with a performance penalty
-     * because it boxes {@code int} into {@code Integer}.<br>
-     * For faster iteration use the {@link #codePointStream()} or {@link #charAt(int)}.
-     *
-     * <p><b>Warning:</b> To iterate over the full contents, you have to also iterate over the strings.
-     *
-     * <p><b>Warning:</b> For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
-     * Do not alter the {@code UnicodeSet} while iterating.
+     * <p><b>Warnings:</b>
+     * <ul>
+     *   <li>This is a convenience method, but comes with a performance penalty
+     *       because it boxes {@code int} into {@code Integer}.<br>
+     *       For for an efficient but old alternative use {@link UnicodeSetIterator#next()}.
+     *   <li>To iterate over the full contents, you have to also iterate over the strings.
+     *   <li>For speed, {@code UnicodeSet} iteration does not check for concurrent modification.<br>
+     *       Do not alter the {@code UnicodeSet} while iterating.
+     * </ul>
      *
      * @return an {@link Iterable} over all the code points
+     *
+     * @draft ICU 76
      */
     public Iterable<Integer> codePoints() {
-        return new CodePointsIterable();
+        return new CodePointIterable(this);
     }
 
-    private class CodePointsIterable implements Iterable<Integer> {
+    private class CodePointIterable implements Iterable<Integer> {
+        private final UnicodeSet unicodeSet;
+
+        CodePointIterable(UnicodeSet unicodeSet) {
+            this.unicodeSet = unicodeSet;
+        }
+
         @Override
         public Iterator<Integer> iterator() {
-            return new CodePointsIterator();
+            return new CodePointsIterator(unicodeSet);
         }
     }
 
     private class CodePointsIterator implements Iterator<Integer> {
```

Comment:
nit: now that CodePointIterable is singular, this should also be singular CodePointIterator

Reply:
Done now.

Sorry, there was another class CodePointsFoo, and that was Done.
Off by one error.